src/Stratis.Bitcoin.Features.Dns/DnsSeedMasterFile.cs

```diff
@@ -46,6 +46,11 @@ public DnsSeedMasterFile(TimeSpan ttl)
         /// </summary>
         public DnsSeedMasterFile() { }
 
+        public DnsSeedMasterFile(IList<IResourceRecord> resourceRecords)
+        {
+            this.entries = resourceRecords;
```

Comment:
I wonder if you should be copying to a new list `resourceRecords.ToList()` so that the same collection is not modified from outside.